Refactor XML profiles [8564] (#150)
eProsima/Fast-RTPS-docs: docs/fastdds/xml_configuration/domainparticipant.rst

|                                   | public data member of the |br|                   |                     |         |
|                                   | |DomainParticipantQosWireProtocolClass|          |                     |         |
|                                   | class. |br|                                      |                     |         |
|                                   | See the                                          |                     |         |
|                                   | :ref:`builtin` section.                          |                     |         |
+-----------------------------------+--------------------------------------------------+---------------------+---------+
| ``<port>``                        | Allows defining the port and gains related |br|  | `Port`_             |         |
|                                   | to the RTPS protocol. See the `Port`_ section.   |                     |         |
+-----------------------------------+--------------------------------------------------+---------------------+---------+

> | ``<participantID>``               | DomainParticipant's identifier. Typically |br|   | ``int32``           | 0       |

Reviewer: Which data member is this?

Author: eprosima::fastrtps::rtps::RTPSParticipantAttributes::participantID

Reviewer 2: Ese cambió de sitio. Es el principal API break

Reviewer 2: Forget it. It was domain id...

|                                   | it will be automatically generated by the |br|   |                     |         |
|                                   | |DomainParticipantFactory|.                      |                     |         |
+-----------------------------------+--------------------------------------------------+---------------------+---------+

> | ``<throughputController>``        | Allows defining a maximum throughput. |br|       | :ref:`Throughput`   |         |

Reviewer: ```suggestion
| ``<throughputController>``        | Limits middleware's bandwidth usage. |br|       | :ref:`Throughput`   |         |
```

|                                   | See the :ref:`Throughput` section.               |                     |         |
+-----------------------------------+--------------------------------------------------+---------------------+---------+

> | ``<userTransports>``              | Transport descriptors to be used by the |br|     | ``List <string>``   |         |

Reviewer: Refer to xml transports

|                                   | DomainParticipant.                               |                     |         |
+-----------------------------------+--------------------------------------------------+---------------------+---------+
| ``<useBuiltinTransports>``        | Boolean field to indicate the system |br|        | ``bool``            | true    |

> |                                   | that the DomainParticipant will use the |br|     |                     |         |
> |                                   | default                                          |                     |         |
> |                                   | |DomainParticipantQosWireProtocolBuiltin|        |                     |         |
> |                                   | transport independently |br|                     |                     |         |

Reviewer: ```suggestion
|                                   | whether the DomainParticipant will use the |br|     |                     |         |
|                                   | default                                          |                     |         |
|                                   | |DomainParticipantQosWireProtocolBuiltin|        |                     |         |
|                                   | transport instead |br|                     |                     |         |
```

|                                   | of its ``<userTransports>``.                     |                     |         |
+-----------------------------------+--------------------------------------------------+---------------------+---------+
| ``<propertiesPolicy>``            | Additional configuration properties. |br|        | |PolicyType|        |         |
|                                   | It expects a |PolicyType|.                       |                     |         |
+-----------------------------------+--------------------------------------------------+---------------------+---------+
| ``<allocation>``                  | Configuration regarding allocation behavior.     | |PartAlloc|         |         |
|                                   | |br| It expects a |br| |PartAlloc|.              |                     |         |
+-----------------------------------+--------------------------------------------------+---------------------+---------+

.. _Port:

Port Configuration
""""""""""""""""""

According to the `RTPS standard <https://www.omg.org/spec/DDSI-RTPS/2.2/PDF>`_ (Section 9.6.1.1), the

> |RTPSParticipants|'discovery traffic unicast listening ports are calculated using the following equation:

Reviewer: ```suggestion
|RTPSParticipants|' discovery traffic unicast listening ports are calculated using the following equation:
```

+------------------------------+------------------------------------------------------+-----------------+--------------+
| ``<offsetd0>``               | Multicast metadata offset.                           | ``uint16``      | 0            |
+------------------------------+------------------------------------------------------+-----------------+--------------+
| ``<offsetd1>``               | Unicast metadata offset.                             | ``uint16``      | 10           |
+------------------------------+------------------------------------------------------+-----------------+--------------+
| ``<offsetd2>``               | Multicast user data offset.                          | ``uint16``      | 1            |
+------------------------------+------------------------------------------------------+-----------------+--------------+
| ``<offsetd3>``               | Unicast user data offset.                            | ``uint16``      | 11           |
+------------------------------+------------------------------------------------------+-----------------+--------------+

> 

Reviewer: Maybe a warning saying that changing this will break compatibility with other RTPS compliant implementations, or even with another app using fast with default ports config

.. _ParticipantAllocationType:

ParticipantAllocationType
"""""""""""""""""""""""""

The ``ParticipantAllocationType`` defines the ``<allocation>`` element, which allows setting of the parameters
related with the allocation behavior on the DomainParticipant.
Please refer to :ref:`participantresourcelimitsqos` for a detailed documentation on DomainParticipants allocation
configuration.

.. list-table::
   :header-rows: 1
   :align: left

   * - Name
     - Description
     - Values
     - Default
   * - ``<max_unicast_locators>``
     - Maximum number of unicast locators expected on a |br|
       remote  entity. It is recommended to use the maximum |br|
       number of network interfaces available on the machine |br|
       on which DomainParticipant is running. |br|
       See :ref:`remotelocatorsallocationattributes`.
     - :ref:`CommonAlloc`
     - 4
   * - ``<max_multicast_locators>``
     - Maximum number of multicast locators expected on a |br|
       remote entity. May be set to zero to disable multicast |br|
       traffic. See :ref:`remotelocatorsallocationattributes`.
     - ``UInt32``

>      - 1

Reviewer: Example shows this inside `<remote_locators>`

    :language: xml
    :start-after: <!-->XML-PARTICIPANT-ALLOCATION<-->
    :end-before: <!--><-->

.. _builtin:

Builtin parameters
"""""""""""""""""""

By calling the |DomainParticipantQosWireProtocol| member function of the |DomainParticipantQos|, it is possible to
access the |DomainParticipantQosWireProtocolBuiltin| public data member of the |DomainParticipantQosWireProtocolClass|
class.
This section specifies the available XML members for the configuration of this
|DomainParticipantQosWireProtocolBuiltin| parameters.

.. Some large words outside of table. Then table fit maximum line length

.. |loclist| replace:: A set of ``<locator>`` |br| members. |br| See :ref:`LocatorListType`
.. |mempol| replace:: :ref:`HistoryMemoryPolicy <memorymanagementpolicy>`
.. |mempoldefault| replace:: :class:`PREALLOCATED_WITH_REALLOC`

+---------------------------------------+--------------------------------------+---------------------+-----------------+
| Name                                  | Description                          | Values              | Default         |
+=======================================+======================================+=====================+=================+

> | ``<discovery_config>``                | This is the main element within |br| |                     |                 |

Reviewer: link to the xml discovery config part. Can use Values cell for that

|                                       | traffic to PDP only.                 |                     |                 |
+---------------------------------------+--------------------------------------+---------------------+-----------------+
| ``<use_WriterLivelinessProtocol>``    | Indicates whether to use the |br|    | ``Boolean``         | :class:`true`   |
|                                       | DataWriterLiveliness protocol.       |                     |                 |
+---------------------------------------+--------------------------------------+---------------------+-----------------+
| ``<metatrafficUnicastLocatorList>``   | Metatraffic Unicast Locator List.    | |loclist|           |                 |
+---------------------------------------+--------------------------------------+---------------------+-----------------+
| ``<metatrafficMulticastLocatorList>`` | Metatraffic Multicast Locator List.  | |loclist|           |                 |
+---------------------------------------+--------------------------------------+---------------------+-----------------+

> | ``<initialPeersList>``                | Initial peers.                       | |loclist|           |                 |

Reviewer: What for?

+---------------------------------------+--------------------------------------+---------------------+-----------------+
| ``<DataWriterHistoryMemoryPolicy>``   | Memory policy for DataWriters. |br|  | |mempol|            | |mempoldefault| |
|                                       | See :ref:`historyqospolicykind`.     |                     |                 |
+---------------------------------------+--------------------------------------+---------------------+-----------------+
| ``<DataReaderPayloadSize>``           | Maximum payload size for |br|        | ``UInt32``          | 512             |
|                                       | DataReaders.                         |                     |                 |
|                                       |                                      |                     |                 |
+---------------------------------------+--------------------------------------+---------------------+-----------------+
| ``<DataWriterPayloadSize>``           | Maximum payload size for |br|        | ``UInt32``          | 512             |

> |                                       | DataWriters.                         |                     |                 |

Reviewer: What does this mean? Does it mean I need to set this to send something bigger than 512 kB? That is not the case

|                                 |                                                | |NONE|              |             |
+---------------------------------+------------------------------------------------+---------------------+-------------+
| ``<ignoreParticipantFlags>``    | Restricts metatraffic using several |br|       | :ref:`partfiltering`| |NO_FILTER| |
|                                 | filtering criteria.                            |                     |             |
|                                 | See :ref:`discovery_ignore_flags`.             |                     |             |
+---------------------------------+------------------------------------------------+---------------------+-------------+
| ``<EDP>``                       | If set to ``SIMPLE``, ``<simpleEDP>`` |br|     | ``SIMPLE``          | ``SIMPLE``  |
|                                 | element would be used. |br|                    +---------------------+             |
|                                 | If set to ``STATIC``, |EDPStatic| will be |br| | ``STATIC``          |             |

> |                                 | performed configured with the contents |br|    |                     |             |

Reviewer: ```suggestion
|                                 | performed, configured with the contents |br|    |                     |             |
```

+--------------------------+-------------------------------------------------------------------------------------------+
| FILTER_DIFFERENT_HOST    | Discovery traffic from another host is discarded.                                         |
+--------------------------+-------------------------------------------------------------------------------------------+
| FILTER_DIFFERENT_PROCESS | Discovery traffic from another process on the same host is discarded.                     |
+--------------------------+-------------------------------------------------------------------------------------------+
| FILTER_SAME_PROCESS      | Discovery traffic from DomainParticipant's own process is discarded.                      |
+--------------------------+-------------------------------------------------------------------------------------------+
| FILTER_DIFFERENT_PROCESS | Discovery traffic from DomainParticipant's own host is discarded.                         |
+--------------------------+                                                                                           |

> | FILTER_SAME_PROCESS      |                                                                                           |

Reviewer: This should be `FILTER_DIFFERENT_PROCESS | FILTER_SAME_PROCESS`. Update also in discovery
